php/ClassInfo/ComposerClassmapFileRepository.php

```diff
@@ -104,13 +100,18 @@ public function find($path_pattern, ClassFilter $filter = null)
      */
     private function isValid($classpath)
     {
-        $filePath = $this->classLoader->findFile($classpath);
```

Comment:
Please note that it is only one of many possible `ClassInfoRepository` repository implementation. You may be confused by two facts: it implementation makes two different thing that should be separated for performance reasons:

- find() implementation should search for pattern through the fast in-memory database and, as now, return a collection of `ClassInfo` objects. Dependently of database implementation the choice of pattern matching may be restricted.
- updating the whole database or reindexing information about single class should be executed only when it is needed or on demand. It is also the place for checking whether if getting information is possible at all (fatal errors mentioned yesterday). How we can gather such information is for our invention (or use external libraries if they exists and do it good). There is the area to deal with different autoloader implementations and manually defined classes.